zerver/tests/test_markdown_thumbnail.py

```diff
@@ -0,0 +1,309 @@
+import re
+from unittest.mock import patch
+
+import pyvips
+
+from zerver.actions.message_delete import do_delete_messages
+from zerver.lib.camo import get_camo_url
+from zerver.lib.markdown import render_message_markdown
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.test_helpers import get_test_image_file
+from zerver.lib.thumbnail import ThumbnailFormat
+from zerver.models import ArchivedAttachment, ArchivedMessage, Attachment, ImageAttachment, Message
+from zerver.models.clients import get_client
+from zerver.worker.thumbnail import ensure_thumbnails
+
+
+class MarkdownThumbnailTest(ZulipTestCase):
+    def upload_image(self, image_name: str) -> str:
+        self.login("othello")
+        with get_test_image_file(image_name) as image_file:
+            response = self.assert_json_success(
+                self.client_post("/json/user_uploads", {"file": image_file})
+            )
+            return re.sub(r"/user_uploads/", "", response["url"])
+
+    def upload_and_thumbnail_image(self, image_name: str) -> str:
+        with self.captureOnCommitCallbacks(execute=True):
+            return self.upload_image(image_name)
```

Comment:
This test helper might benefit from a comment reminding a reader why this causes thumbnailing to occur.